File: apps/desktop/src/types/study.ts
// Study types

export interface StudySubject {
  id: number;
  name: string;
  description: string | null;
  priority: string;
  progress: number;
  weekly_goal_hours: number;
  monthly_goal_hours: number;
  color: string;
  created_at: string;
  updated_at: string;
}

export interface SubjectCreate {
  name: string;
  description?: string;
  priority?: string;
  weekly_goal_hours?: number;
  monthly_goal_hours?: number;
  color?: string;
}

export interface SubjectUpdate {
  name?: string;
  description?: string;
  priority?: string;
  progress?: number;
  weekly_goal_hours?: number;
  monthly_goal_hours?: number;
  color?: string;
}

export type StudyMode = "pomodoro" | "free";

export interface StudySession {
  id: number;
  subject_id: number;
  subject_name: string;
  mode: StudyMode;
  started_at: string;
  ended_at: string | null;
  duration_minutes: number;
  notes: string | null;
  created_at: string;
}

export interface SessionCreate {
  subject_id: number;
  mode: StudyMode;
  started_at: string;
  ended_at?: string;
  duration_minutes: number;
  notes?: string;
}

export interface StudySummary {
  today_minutes: number;
  week_minutes: number;
  month_minutes: number;
  daily_goal_minutes: number;
  weekly_goal_minutes: number;
  streak_days: number;
}

File: apps/backend/app/schemas/study.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List
from datetime import datetime
from app.models.study import SessionMode

class SubjectCreate(BaseModel):
    name: str
    description: Optional[str] = None
    priority: int = 1
    weekly_goal_hours: Optional[float] = None
    monthly_goal_hours: Optional[float] = None
    color: Optional[str] = None

class SubjectUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    priority: Optional[int] = None
    progress: Optional[float] = None
    weekly_goal_hours: Optional[float] = None
    monthly_goal_hours: Optional[float] = None
    color: Optional[str] = None

class SubjectResponse(BaseModel):
    id: int
    name: str
    description: Optional[str]
    priority: int
    progress: float
    weekly_goal_hours: Optional[float]
    monthly_goal_hours: Optional[float]
    color: Optional[str]
    created_at: datetime
    
    model_config = ConfigDict(from_attributes=True)

class SessionCreate(BaseModel):
    subject_id: int
    mode: SessionMode
    started_at: datetime
    ended_at: Optional[datetime] = None
    duration_minutes: Optional[int] = None
    notes: Optional[str] = None

class SessionResponse(BaseModel):
    id: int
    subject_id: int
    mode: SessionMode
    started_at: datetime
    ended_at: Optional[datetime]
    duration_minutes: Optional[int]
    notes: Optional[str]
    
    model_config = ConfigDict(from_attributes=True)

class StudySummary(BaseModel):
    hours_today: float
    hours_this_week: float
    hours_this_month: float
    by_subject: List[dict]
